control/controlclient/map.go

```diff
@@ -828,6 +829,17 @@ func (ms *mapSession) sortedPeers() []tailcfg.NodeView {
 func (ms *mapSession) netmap() *netmap.NetworkMap {
 	peerViews := ms.sortedPeers()
 
+	var displayMessages []tailcfg.DisplayMessage
+	for _, h := range ms.lastHealth {
+		m := tailcfg.DisplayMessage{
+			ID:       "control-health",
```

Comment:
kinda weird having N distinct messages all sharing the same ID.

to me, an ID means it's unique

Reply:
It is kinda weird I agree. The API here between the Go code and the UI is a map from warnable code to unhealthy state:

```
type State struct {
	Warnings map[WarnableCode]UnhealthyState
}
```

So ultimately only one warnable per unique code will be visible to the client. That's rather unfortunate as the API between the Go code and the control plane is a list of strings.

I checked and the current behaviour is that if multiple control health messages are in the netmap, the _last_ message is the one displayed. This code does match the current behaviour... but perhaps it could make it more explicit? Alternatively we could generate new IDs, but I'm not sure it's worth the effort as we a) don't use these messages often and b) are replacing them with the v2 version anyway.

Reply:
As of dfbb75d0415c68d2de858b15089dcc21e727e886, all control health messages now show up in CurrentState().Warnings